### PR TITLE
Add Gaussian support

### PR DESCRIPTION
**Pull request by jthorton**
This PR adds gaussian as an engine to geomeTRIC. Both gaussian 16 and 09 input files are supported in cartesian coordinates only. Tests and examples are also added. Please let me know if I have missed anything or if we need more examples.

### CODE REVIEW

**Comment by jthorton** [comment]
@leeping I think I have made all of the changes we have discussed, we can now set the number of processors to use from the command line even if the user has supplied a value in the input file. The checkpoint name is also always overwritten to make sure it is consistent and I have added some more examples including the use of an implicit solvent with settings at the bottom of the file. The regex string also seems to be working as we expect when reading the Gaussian input files. Let me know if you spot anything else.

File: geometric/prepare.py [hunk 1]
```diff
@@ -119,7 +120,7 @@ def get_molecule_engine(**kwargs):
     if engine_str:
         engine_str = engine_str.lower()
         if engine_str[:4] == 'tera' : engine_str = 'tera'
-        if engine_str not in ['tera', 'qchem', 'psi4', 'gmx', 'molpro', 'openmm', 'qcengine']:
+        if engine_str not in ['tera', 'qchem', 'psi4', 'gmx', 'molpro', 'openmm', 'qcengine', "gaussian"]:
             raise RuntimeError("Valid values of engine are: tera, qchem, psi4, gmx, molpro, openmm, qcengine")
```
Comment (leeping) [inline geometric/prepare.py:124]: Could you please add gaussian to this error message?

File: examples/solvent_gaussian/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Here we run an optimization with an implicit solvent.
```
Comment (leeping) [inline examples/solvent_gaussian/README.md:1]: Could you rename this folder to `ethane_pcm_gaussian`?

File: geometric/tests/test_gaussian.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        molecule_2 = Molecule(os.path.join("ethane.tmp", "gaussian.com"))
+        # now check over the data
+        assert molecule.Data["elem"] == molecule_2.Data["elem"]
+        assert molecule.Data["bonds"] == molecule_2.Data["bonds"]
+        assert np.allclose(molecule.Data["xyzs"][0], molecule_2.Data["xyzs"][0])
+
+        os.chdir(home)
+
+
+def test_read_results_gaussain():
```
Comment (leeping) [inline geometric/tests/test_gaussian.py:153]: Spelling of "Gaussian"

File: geometric/tests/test_gaussian.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    if not int(major) >= 3 and int(minor) >= 5:
+        pytest.skip("Python version below 3.5 TemporaryDirectory not available.")
+    molecule = Molecule(os.path.join(datad, "ethane.com"))
+    engine = Gaussian(molecule=molecule, exe="g09")
+    engine.load_gaussian_input(os.path.join(datad, "ethane.com"))
+    home = os.getcwd()
+    with tempfile.TemporaryDirectory() as temp:
+        os.chdir(temp)
+        # now we want to run calc new to make sure the file is written correctly
+        with pytest.raises(GaussianEngineError):
```
Comment (leeping) [inline geometric/tests/test_gaussian.py:140]: What happens if `engine.calc()` does not throw an error (for example if Gaussian really does exist on the machine running the test?)

**Review by leeping — CHANGES_REQUESTED** [review]

File: geometric/tests/test_gaussian.py (hunk 4 above)
Comment (jthorton) [reply geometric/tests/test_gaussian.py:140]: Here we now allow the test to run if gaussian is installed.

**Review by leeping — APPROVED** [review]
Thank you, this is excellent. :) LGTM

**Pull request merged**